### PR TITLE
Minor refactorings

### PR DESCRIPTION
**Pull request by gzsombor**
Small changes, which I thought make sense.

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/module/domain/properties/JHipsterServerPort.java [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
 
-  private final int serverPort;
+  public JHipsterServerPort {
+    Assert.field("serverPort", serverPort).min(1).max(65_535);
+  }
 
   public JHipsterServerPort(Integer port) {
-    serverPort = buildPort(port);
-
-    Assert.field("serverPort", serverPort).min(1).max(65_535);
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/module/domain/properties/JHipsterServerPort.java:14]: Why are you removing this assertion?

File: src/main/resources/generator/buildtool/maven/pom.xml.mustache [hunk 2]
```diff
@@ -14,7 +14,7 @@
   <packaging>jar</packaging>
 
   <properties>
-    <java.version>21</java.version>
+    <java.version>{{javaVersion}}</java.version>
```
Comment (murdos) [inline src/main/resources/generator/buildtool/maven/pom.xml.mustache:17]: What is your intent? 
Generated code only compiles with java 21+
By allowing user to change this property to any value, there's great chance that the generated project doesn't work.

**Review by murdos — CHANGES_REQUESTED** [review]

File: src/main/java/tech/jhipster/lite/module/domain/properties/JHipsterServerPort.java (hunk 1 above)
Comment (gzsombor) [reply src/main/java/tech/jhipster/lite/module/domain/properties/JHipsterServerPort.java:14]: The assertion is moved into the default constructor, so it's still apply

File: src/main/resources/generator/buildtool/maven/pom.xml.mustache (hunk 2 above)
Comment (gzsombor) [reply src/main/resources/generator/buildtool/maven/pom.xml.mustache:17]: The user can't change the property, it is set in the JHipsterModuleProperties as a final variable.
I don't want to repeat the same 21 in every template, and make it clear what's that number is about.
So I can write a template, without the need to constantly update, whenever a new java version is released.

**Review by murdos — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Happy to see you here @gzsombor !
I didn't expect you followed this project. Thanks for your help and your contribution :)

**Pull request merged**